test/suites/integration/backend/upload-download-file.test.ts

```diff
@@ -268,5 +272,36 @@ await describeMspNet(
         "Root of bucket should have changed"
       );
     });
+
+    it("Should successfully download a file via the backend API", async () => {
+      // Ensure the upload test completed successfully
+      assert(uploadedFileKeyHex, "Upload test must complete successfully before download test");
+      assert(originalFileBuffer, "Original file buffer must be available from upload test");
+
+      const response = await fetch(`http://localhost:8080/download/${uploadedFileKeyHex}`);
+      strictEqual(response.status, 200, "Download endpoint should return 200 OK");
+
+      const contentDisposition = response.headers.get("content-disposition");
+      assert(contentDisposition, "Content disposition should be present");
+      // Filename is preserved from the upload request
+      strictEqual(
+        contentDisposition,
+        'attachment; filename="whatsup.jpg"',
```

Comment:
can we have that `whatsup.jpg` be also a variable like the file key? if ever we change the file name it should be clear that it's a dependency